### ISSUE TITLE
A lot of repeated json and plaintext tests

### ISSUE THREAD

**Issue by joanhey** [TechEmpower/FrameworkBenchmarks#8178]
After check it, a lot of frameworks are **repeating json and plaintext** tests.
Some are for different platforms or libs, and it's OK.
**But many only because have them in database variants.**

**Plaintext repeated:**

- [ ]  Sinatra **14**
- [ ]  Officefloor  **11** @sagenschneider 
- [ ]  Falcon  **11** @remittor 
- [ ]  Fastapi **11** @micaelmalta 
- [ ]  Chi  **10** @0uep 
- [ ]  Flask  **7** @remittor 
- [ ]  Roda  **6**  @timuckun  #8160
- [ ]  Nestjs **6** @JHyeok 
- [ ]  ...

Work in progress.

**Comment by remittor** [comment]
![image](https://user-images.githubusercontent.com/9350115/234288762-f3b2d446-6219-419e-96ac-f262550d45c4.png)

@joanhey , do I need to remove only test `falcon [meinheld-orjson]` ?
All other tests have different http servers.

**Comment by remittor** [comment]
![image](https://user-images.githubusercontent.com/9350115/234290188-8ae140a9-1e76-4d9d-a6aa-f8a761f922a8.png)

@joanhey , `PyPy` variants (`flask-raw` & `flask`) already marked as `broken` : https://github.com/TechEmpower/FrameworkBenchmarks/pull/8169/commits/f8e8239a242fa57a40c727ab25bcfcac199afee0#diff-bf7a33c9cd47aa475c2802732158ab4d7009ed11dafefea8a6114204558925d3R112

**Comment by joanhey** [comment]
@remittor orjson looks like a different json lib, but the plaintext and fortunes don't use json.
 
About flask:
![image](https://user-images.githubusercontent.com/249085/234306188-d042b8a7-61f1-4b3f-81f3-b905a75fe88a.png)

![image](https://user-images.githubusercontent.com/249085/234306562-50ba56d5-a254-4757-a3da-f8b07c508f41.png)

**Comment by joanhey** [comment]
I don't know the difference between Sinatra and Sinatra Sequel.

But in the plaintext test, the results are identical.

![image](https://user-images.githubusercontent.com/249085/234402305-70252d76-b451-46dd-9f71-47a6a65931e4.png)
![image](https://user-images.githubusercontent.com/249085/234402414-20ef4112-6c29-4e90-80b1-0f5e57fb3d3f.png)

**Comment by timuckun** [comment]
The difference between Sinatra and Sinatra Sequel is the ORM. Sinatra uses ActiveRecord and Sinatra Sequel uses sequel.

Of course neither the JSON nor the plaintext tests touch the database so it doesn't matter in those cases.

One issue with the ruby benchmarks are that they all test mysql and postges and three different servers (puma, unicorn, passenger).  Every combination is a different performance profile but I think the puma-postgres combination is likely to be fastest. 

Also the ruby tests are quite old (except for rails). I just did a pull request for Roda to update it and will also update the other ones soon. This should make some difference in the performance of the ruby benchmarks.

**Comment by joanhey** [comment]
@timuckun remember to delete the extra tests in Roda
Thank you

**Comment by timuckun** [comment]
Which are the extra tests? Do you mean the MySQL tests or the the other servers, or the plaintext/json tests?

It seems to me there is some value in knowing which of the various ruby servers performs best under these loads. 

Perhaps a good idea is to test the servers in the currently broken rack tests and then use the fastest one in the rest of the ruby tests. Rack is the underlying library in virtually all the ruby frameworks so that would establish a good baseline for testing various threading/fiber/forking/async tests.

**Comment by micaelmalta** [comment]
Indeed, the web-server chosen will respond differently against the latency of the application.
So, IMO, it s a good idea to the test all the possible combinations

**Comment by joanhey** [comment]
If you test a Json lib, you don't need Plaintext and Fortunes, as don't use json.

For every server is OK to have all the tests.
But with the same server and different database, we don't need to repeat the Json and Plaintext.

**Comment by timuckun** [comment]
The way ruby works is that you use the same class for the functionality and use a different invocation based on the server.

For example 

```bash
bundle exec passenger start ....
bundle exec unicorn ...
bundle exec puma ...
```

All of these will all read the same config.ru file which will execute whatever class is defined in there. Each docker file is slightly different but the code is the same for all of the servers.  The path of least resistance is the repeat all the tests for each server.

I am happy to get rid of the mysql tests but in the cast of ORMs it might be useful to see how the performance of the ORM and the client libs differ between the databases.

As mentioned  above here is what I propose.

I would fix the broken ruby/rack tests and test all the servers in that  project and then limit the other ruby tests to whichever server is the fastest.  Perhaps the rack test could also test both databases using native libs. This would establish a baseline for all ruby based benchmarks as they all depend on rack. Once that baseline is established each framework would only test their own project with one server and one database using whatever ORM or library it comes bundled with.  I don't think every server needs to be tested with both databases so I would use the most widely used server (puma) to test both databases the rest would only be tested with postgres.

**Comment by joanhey** [comment]
Of course that you need to test for every server and database.

But the same server, don't need to test json and plaintext for each database.
We only need to test the database tests.
The json and plaintext will be the same with any database, for that we don't need to repeated.

Example plaintext:
![image](https://user-images.githubusercontent.com/249085/234418716-f0403ff7-3428-4228-ac73-4ccc6aa2b362.png)

The passenger mri, will have **the same results in json and plaintext** with any database. 
So we only need to test it one time, and not with any database.

**Comment by timuckun** [comment]
I understand what you are saying. I can remove it from the benchmarks.json for the additional tests.

**Comment by KostyaTretyak** [comment]
In the directory `frameworks/JavaScript` and `frameworks/TypeScript` there are often unnecessary test duplicates. For example, this applies to `nodejs`, `fastify`, etc.

**Comment by JHyeok** [comment]
@KostyaTretyak 
I don't see any repeated tests in `nodejs`, `fastify`, and `nest` except for chakra.

**Comment by KostyaTretyak** [comment]
If you see a mix, for example, `plaintext_url` with `db_url` - these are duplicates. And there are indeed many, including [nodejs][nodeurl], [fastify][fastifyurl], [nest][nesturl] and so on...

It makes no sense to make such a mix, since the results should contain statistics plaintext with json or databases.


[nodeurl]: https://github.com/TechEmpower/FrameworkBenchmarks/blob/270ca5e6f940ae5c426bffb07caf01ce9f9eb62c/frameworks/JavaScript/nodejs/benchmark_config.json#L6-L7
[fastifyurl]: https://github.com/TechEmpower/FrameworkBenchmarks/blob/270ca5e6f940ae5c426bffb07caf01ce9f9eb62c/frameworks/JavaScript/fastify/benchmark_config.json#L7-L8
[nesturl]: https://github.com/TechEmpower/FrameworkBenchmarks/blob/270ca5e6f940ae5c426bffb07caf01ce9f9eb62c/frameworks/TypeScript/nest/benchmark_config.json#L7-L8

**Comment by NateBrady23** [comment]
Closing as part of bulk cleanup.
